mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/ConvertScreenV2.tsx

```diff
@@ -0,0 +1,401 @@
+import { AddressToken } from '@defichain/whale-api-client/dist/api/address'
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { View } from '@components'
+import {
+  ThemedIcon,
+  ThemedScrollViewV2,
+  ThemedSectionTitleV2,
+  ThemedTextV2,
+  ThemedTouchableOpacityV2,
+  ThemedViewV2
+} from '@components/themed'
+import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
+import { RootState } from '@store'
+import { hasTxQueued as hasBroadcastQueued } from '@store/ocean'
+import { hasTxQueued } from '@store/transaction_queue'
+import { tailwind } from '@tailwind'
+import { translate } from '@translations'
+import { PortfolioParamList } from '../PortfolioNavigator'
+import { useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { tokensSelector } from '@store/wallet'
+import { getNativeIcon } from '@components/icons/assets'
+import { ButtonV2 } from '@components/ButtonV2'
+import { AmountButtonTypes, TransactionCard } from '@components/TransactionCard'
+import { useToast } from 'react-native-toast-notifications'
+import NumberFormat from 'react-number-format'
+import { WalletTransactionCardTextInput } from '@components/WalletTransactionCardTextInput'
+
+export type ConversionMode = 'utxosToAccount' | 'accountToUtxos'
+type Props = StackScreenProps<PortfolioParamList, 'ConvertScreen'>
+
+interface ConversionIO extends AddressToken {
+  unit: 'UTXO' | 'Token'
+}
+
+export function ConvertScreenV2 (props: Props): JSX.Element {
+  const client = useWhaleApiClient()
+  const logger = useLogger()
+  const tokens = useSelector((state: RootState) => tokensSelector(state.wallet))
+  const toast = useToast()
+  const TOAST_DURATION = 2000
+
+  // global state
+  const hasPendingJob = useSelector((state: RootState) => hasTxQueued(state.transactionQueue))
+  const hasPendingBroadcastJob = useSelector((state: RootState) => hasBroadcastQueued(state.ocean))
+  const navigation = useNavigation<NavigationProp<PortfolioParamList>>()
+  const [mode, setMode] = useState(props.route.params.mode)
+  const [sourceToken, setSourceToken] = useState<ConversionIO>()
+  const [targetToken, setTargetToken] = useState<ConversionIO>()
+  const [convAmount, setConvAmount] = useState<string>('0')
+  const [fee, setFee] = useState<BigNumber>(new BigNumber(0.0001))
+  const [amount, setAmount] = useState<string>('')
+  const [hasError, setHasError] = useState<boolean>(false)
+  const [showMaxUTXOWarning, setShowMaxUTXOWarning] = useState<boolean>(false)
+  const [transactionCardStatus, setTransactionCardStatus] = useState<'error' | 'active' | undefined>()
```

Comment:
Can we use enum here instead?

Reply:
done, use enum for inline text status and transaction card border status